samples/matter/lock/CMakeLists.txt

```diff
@@ -89,7 +90,12 @@ endif()
 
 if(CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS)
     target_sources(app PRIVATE ${COMMON_ROOT}/src/diagnostic/diagnostic_logs_provider.cpp)
+if(CONFIG_NCS_SAMPLE_MATTER_DIAGNOSTIC_LOGS_CRASH_LOGS)
+    # Wrap z_fatal_error to allow injecting crash data into the retention memory.
+    target_link_options(app INTERFACE -Wl,--wrap=z_fatal_error)
```

Comment:
This should be eventually moved to the to-be-created CMakeLists.txt for samples/matter/common.

Reply:
Yes, I agree. I would like to do it together with all other entries just after completing a task for moving all common sources to the `samples/matter/common/` :)